src/mlpack/core/data/encoding_policies/dictionary_encoding.hpp

```diff
@@ -53,21 +53,19 @@ class DictionaryEncoding
   * @param col The column at which the encoding belongs to.
   */
   template<typename MatType>
-  static void Encode(std::unordered_map<boost::string_view, size_t,
-                     boost::hash<boost::string_view>>& mappings,
-                     std::deque<std::string>& /*originalStrings*/,
-                     std::vector< std::vector<boost::string_view> >& dataset,
-                     MatType& output)
+  static void Encode(size_t ele, MatType& output, size_t row, size_t col)
   {
-    for (size_t i = 0; i < dataset.size(); ++i)
-      for (size_t j = 0; j < dataset[i].size(); ++j)
-        output.at(i, j) = mappings.at(dataset[i][j]);
+    output.at(row, col) = ele;
   }
 }; // class DicitonaryEncoding
 
-} // namespace data
-} // namespace mlpack
+template<>
+struct PolicyTraits<DictionaryEncoding> {
 
+  static const bool outputWithNoPadding = true;
```

Comment:
Unfortunately, it's not that easy:) You can't redefine only one property, you have to redefine others as well.